pkg/virt-launcher/virtwrap/cmd-server/server.go

```diff
@@ -310,7 +310,10 @@ func (l *Launcher) GetDomain(_ context.Context, _ *cmdv1.EmptyRequest) (*cmdv1.D
 	if len(list) > 0 {
 		domainObj := list[0]
 		domainObj.Status.OSInfo = l.domainManager.GetGuestOSInfo()
-		domainObj.Status.Interfaces = l.domainManager.InterfacesStatus(domainObj.Spec.Devices.Interfaces)
+		interfaces := l.domainManager.InterfacesStatus(domainObj.Spec.Devices.Interfaces)
+		if interfaces != nil {
```

Comment:
I would add a unit test here to make sure that if `l.domainManager.InterfacesStatus(domainObj.Spec.Devices.Interfaces)` is `nil` the returned `domainObj.Status.Interfaces` is nil.

Reply:
It should be left untouched, not nil
(in case it was nil it will of course stay nil)

it depends on the redesign, i will try to create a commit for it

Reply:
We want to test the no guest agent scenario, so I would test it with `nil`. But of course the unit test for other values are welcome as well.